packages/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/SnapshotUtils.h

```diff
@@ -0,0 +1,159 @@
+
+#include "LayoutAnimationsManager.h"
+#include "PropsRegistry.h"
+
+#include <react/renderer/mounting/MountingOverrideDelegate.h>
+#include <react/renderer/mounting/ShadowView.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#ifdef __ANDROID__ // parse color on android
+#include <iomanip>
+#include <sstream>
+#endif
+
+namespace reanimated {
+
+#ifdef __ANDROID__
+static inline std::string colorToString(const SharedColor &color) {
+  // Operator "*" of class SharedColor is overloaded and returns a private
+  // class member color_ of type Color
+  int val = *color;
+
+  std::stringstream
+      invertedHexColorStream; // By default transparency is first, color second
+  invertedHexColorStream << std::setfill('0') << std::setw(8) << std::hex
+                         << val;
+
+  auto invertedHexColor = invertedHexColorStream.str();
+  auto hexColor =
+      "#" + invertedHexColor.substr(2, 6) + invertedHexColor.substr(0, 2);
+
+  return hexColor;
+}
+
+#else
+inline std::string colorToString(const SharedColor &value) {
+  ColorComponents components = colorComponentsFromColor(value);
+  auto ratio = 255.f;
+  return "rgba(" + folly::to<std::string>(round(components.red * ratio)) +
+      ", " + folly::to<std::string>(round(components.green * ratio)) + ", " +
+      folly::to<std::string>(round(components.blue * ratio)) + ", " +
+      folly::to<std::string>(round(components.alpha * ratio)) + ")";
+}
+#endif
+
+inline Float floatFromYogaOptionalFloat(yoga::FloatOptional value) {
+  if (value.isUndefined() || std::isnan(value.unwrap())) {
+    return 0;
+  }
+  return (Float)value.unwrap();
+}
+
+inline Float floatFromYogaBorderWidthValue(const yoga::Style::Length &length) {
+  if (length.unit() == yoga::Unit::Point) {
+    return floatFromYogaOptionalFloat(length.value());
+  }
+  // types percent, and auto are not allowed for border width
+  return 0;
+}
+
+struct Snapshot {
+  double x, y, width, height, windowWidth, windowHeight;
+  Snapshot(const ShadowView &shadowView, Rect window) {
+    const auto &frame = shadowView.layoutMetrics.frame;
+    x = frame.origin.x;
+    y = frame.origin.y;
+    width = frame.size.width;
+    height = frame.size.height;
+    windowWidth = window.width;
+    windowHeight = window.height;
+  }
+};
+
+static const int numberOfStringProperties = 6;
+static const char *stringPropertiesNames[6] = {
+    "BackgroundColor",
+    "ShadowColor",
+
+    "BorderTopColor",
+    "BorderRightColor",
+    "BorderBottomColor",
+    "BorderLeftColor",
+};
+
+static const int numberOfNumericProperties = 9;
+static const char *numericPropertiesNames[9] = {
+    "Opacity", // double
+
+    "BorderTopLeftRadius", // int
+    "BorderTopRightRadius",
+    "BorderBottomLeftRadius",
+    "BorderBottomRightRadius",
+
+    "BorderLeftWidth", // float
+    "BorderRightWidth",
+    "BorderTopWidth",
+    "BorderBottomWidth"};
+
+struct StyleSnapshot {
```

Comment:
The names Snapshot and StyleSnapshot might be confusing. Could you come up with different naming?

Reply:
Let's have `LayoutSnapshot` and `StyleSnapshot`